### ISSUE TITLE
Ask for Motion & Fitness via requestPermission() on ios

### ISSUE THREAD

**Issue by Vasek18** [transistorsoft/cordova-background-geolocation-lt#1301]
## Your Environment
* Plugin version:
* Platform: iOS or Android
* OS version:
* Device manufacturer / model:
* Cordova version (`cordova -v`):
* Cordova platform version (`cordova platform ls`):
* Plugin config provided to `#ready`:
```javascript <!-- syntax-highlighting:  paste your code below -->
{
    activityRecognitionInterval = 10000;
    activityType = 1;
    authorization = {
    };
    autoSync = 1;
    autoSyncThreshold = 0;
    batchSync = 1;
    debug = 0;
    desiredAccuracy = "-1";
    desiredOdometerAccuracy = 100;
    didLaunchInBackground = 0;
    didRequestUpgradeLocationAuthorization = 1;
    disableAutoSyncOnCellular = 0;
    disableElasticity = 0;
    disableLocationAuthorizationAlert = 0;
    disableMotionActivityUpdates = 0;
    disableStopDetection = 0;
    distanceFilter = 100;
    elasticityMultiplier = 1;
    enableTimestampMeta = 0;
    enabled = 0;
    extras = {
        platform = ios;
    };
    geofenceInitialTriggerEntry = 1;
    geofenceProximityRadius = 2000;
    geofenceTemplate = "";
    headers = {
        Authorization = "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2MzA5Mjg4NDUsImV4cCI6MTYzMTUzMzY0NSwicm9sZXMiOlsiUk9MRV9BRE1JTiJdLCJpZCI6Mzc0LCJhY2NvdW50SWQiOjIwOSwic3NvUHJvdmlkZXIiOiJ6ZW5kZXNrIn0.LAONC9Rk3YI24rNEXS5MFD6WyYt2wV0G1Gmnp34qK9SdkVBOHRteRjfxvrtIODz_b3xN8vThKiphYF-KL4JU_DdOe-G44JOmIlLgKpGob5tbuznpTyhJIWBlU1b99mmaPG-v8FdjMftLquboJcFnXKyJe0IkPSExG3jTMcmYrjWA9iAuAl30kaSu1DC3JcPrNL2tYesxrAXbQX2LkRq_2Jr8yT6N99m5wwMUaVJyEjOLexZBssUUZ1-35yoTT0KDF4TskEI9dd-mZ9ofagxQ-fT6SKB4PzY2agdolp2dUVp7of2A6bk3f41rhGYwHOML9D7ZkE3K6kLmuA9NANFJ3JAcSaNgOdIdyscJRXuAYUwq9IGocUzsaVem5RyeduXuSBGWMw4oZGxKvbp9U0mwsKdED-XIrjtY-A_dqltn-8XQD_adjrEQ3DheBl1hIfaPfMpfH9O2RSiAA-Lpvq1An3MoQ2v7ACJMK3i5wUiuEJuQn4b01vy8A0rtx5NWZgpMdoBCTTU44Ewse8dxvFUOkyQ3QF-tgSIvjQJxojo1eqkMO0Bg4eF1Oz7zgx7whxdl2nWEx1hTRu7R30d1qw0wHXwTNRKx0z5qGRqzu8TMNG_cEYKGPZ6M1tFFXRswCX_XsFd7SV3cgvaDWjtaBdaD_-YxBGumfQrsg8j83NVsjD4";
    };
    heartbeatInterval = 120;
    httpRootProperty = locations;
    httpTimeout = 60000;
    iOSHasWarnedLocationServicesOff = 1;
    isFirstBoot = 1;
    isMoving = 0;
    lastLocationAuthorizationStatus = 3;
    locationAuthorizationAlert = {
        cancelButton = Cancel;
        instructions = "To use background location, you must enable '{locationAuthorizationRequest}' in the Location Services settings";
        settingsButton = Settings;
        titleWhenNotEnabled = "Background location is not enabled";
        titleWhenOff = "Location services are off";
    };
    locationAuthorizationRequest = Always;
    locationTemplate = "";
    locationTimeout = 60;
    locationsOrderDirection = ASC;
    logLevel = 5;
    logMaxDays = 3;
    maxBatchSize = 20;
    maxDaysToPersist = 3;
    maxRecordsToPersist = "-1";
    method = POST;
    minimumActivityRecognitionConfidence = 70;
    odometer = 0;
    params = {
    };
    pausesLocationUpdatesAutomatically = 1;
    persistMode = 2;
    preventSuspend = 1;
    schedule = (
        "2 15:52-19:00",
        "6 9:00-18:00",
        "1 9:00-18:00",
        "3 9:00-18:01",
        "5 7:01-22:32",
        "7 9:00-18:00",
        "4 9:00-19:12"
    );
    schedulerEnabled = 1;
    showsBackgroundLocationIndicator = 1;
    startOnBoot = 1;
    stationaryRadius = 50;
    stopAfterElapsedMinutes = "-1";
    stopDetectionDelay = 0;
    stopOnStationary = 0;
    stopOnTerminate = 0;
    stopTimeout = 2;
    trackingMode = 1;
    url = "https://api.mapsly.com/location_tracking/save";
    useSignificantChangesOnly = 0;
}
```

## Expected Behavior
If we call the _requestPermission()_ method a user among other dialogues see the dialog about "Motion & Fitness"

## Actual Behavior
If we call the _requestPermission()_ method the "Motion & Fitness" permission doesn't get asked. User see the dialog about "Motion & Fitness" only after a call of the _start()_ method

## Steps to Reproduce

1. Call _ready()_
2. Run code
 
 ```
BackgroundGeolocation.setConfig({locationAuthorizationRequest: 'Always'}).then(async () => {
    BackgroundGeolocation.requestPermission();
});
```

## Context
Also we now investigating unexpected tracking stops without any logs. Can lack of the Motion & Fitness permission be the reason that there is no bg tracking?

## Debug logs

<details>
	<summary>Logs</summary>

```<!-- Syntax highlighting:  DO NOT REMOVE -->
2021-09-06 14:46:59.267 🔵+[LocationAuthorization run:onCancel:] Request: requestAlwaysAuthorization

2021-09-06 14:46:59.396 🔴-[LocationManager stopWatchPosition] STOP

2021-09-06 14:47:00.616 ℹ️-[TSLocationManager log:message:] [bg geo] User token update

2021-09-06 14:47:00.618 ℹ️-[TSConfig persist] 

2021-09-06 14:47:00.668 ℹ️-[TSLocationManager log:message:] [bg geo] User token update

2021-09-06 14:47:00.669 ℹ️-[TSConfig persist] 

2021-09-06 14:47:01.756 ℹ️-[TSConfig persist] 

2021-09-06 14:47:01.758 🔵-[TSLocationManager locationManager:didChangeAuthorizationStatus:] status 4

2021-09-06 14:47:01.761 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 4

2021-09-06 14:47:01.761 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 4

2021-09-06 14:47:01.761 🔵-[LocationManager locationManager:didChangeAuthorizationStatus:] 4

2021-09-06 14:47:01.762 🔵-[BackgroundTaskManager locationManager:didChangeAuthorizationStatus:] 4

2021-09-06 14:47:01.763 ℹ️+[LocationAuthorization run:onCancel:] status: 4

2021-09-06 14:47:01.763 ℹ️-[TSConfig persist] 

2021-09-06 14:47:01.765 ℹ️+[LocationAuthorization run:onCancel:] Location authorization upgrade:  WhenInUse -> Always

2021-09-06 14:47:01.765 🔵+[LocationAuthorization run:onCancel:] Request: requestAlwaysAuthorization

2021-09-06 14:47:01.773 🔵-[TSLocationManager getCurrentPosition:] 

2021-09-06 14:47:01.773 🎾-[LocationManager startUpdatingLocation] ON

2021-09-06 14:47:01.778 ℹ️+[LocationAuthorization run:onCancel:] status: 4

2021-09-06 14:47:01.778 ℹ️-[TSConfig persist] 
```

</details>

**Comment by christocracy** [comment]
> Can lack of the Motion & Fitness permission be the reason that there is no bg tracking?

No.

**Comment by Vasek18** [comment]
Thanks for answer
But what about requestPermission() and Motion & Fitness?

**Comment by Vasek18** [comment]
Hi,

We run into a case with no logs on iphone (iPhone XR, 14.7.1). I've attached logs as file. Here is config:
{
    activityRecognitionInterval = 10000;
    activityType = 1;
    authorization = {
    };
    autoSync = 1;
    autoSyncThreshold = 0;
    batchSync = 1;
    debug = 0;
    desiredAccuracy = "-1";
    desiredOdometerAccuracy = 100;
    didLaunchInBackground = 0;
    didRequestUpgradeLocationAuthorization = 0;
    disableAutoSyncOnCellular = 0;
    disableElasticity = 0;
    disableLocationAuthorizationAlert = 0;
    disableMotionActivityUpdates = 0;
    disableStopDetection = 0;
    distanceFilter = 100;
    elasticityMultiplier = 1;
    enableTimestampMeta = 0;
    enabled = 1;
    extras = {
        platform = ios;
    };
    geofenceInitialTriggerEntry = 1;
    geofenceProximityRadius = 2000;
    geofenceTemplate = "";
    headers = {
        Authorization = "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2MzE3MDI0MDMsImV4cCI6MTYzMjMwNzIwMywicm9sZXMiOlsiUk9MRV9BRE1JTiJdLCJpZCI6MTE3NSwiYWNjb3VudElkIjo1OTcsInNzb1Byb3ZpZGVyIjpudWxsfQ.KNKa7WghJd5_BnF_mAPSOzxu7-172X33U4LysSlpN01aK_-qv0aLlSrFn_dRaetWRBRZIx7nh5wimxBoNG2VzjZ6IbncMg09vAY6vCi9K0kwYf6MHkgVVVpdGmh2oVY1gIXSDGjVhToqzn94vyjV-ou1psJB_G1pga4shSuRJZ5t7VJJpNaLdLgu7YtbppE9U5wAUrdzJwDt-cPiULMdz8TWMLkvFFdbs8eUY91ytfiXDCUP_2aYbv5YWl2S0_0yHhiM8C3MyGsBPh-rcjwtScpUdG__IO556rrDifmVy4_ztiAh1khHEW92e9P978E15E8fV2M8tBdghxD1kVg9qbnLyI-6hpJK7sl2nWUA1HKoYacufNx39lw-TeVNE8WodnlmBSUGLG3ol_P-GkK21XrI_dwkyEaNMc6Ubf5UVolgYPUdYvGSKwdHVIO-MQZ8HWREpUgzPcWQT7T-JBeKOjZg8gw4f74JPxxQnDFjOzg6-eWcjJS-yGphnyNMBU5OAXH2U2c1frzhM0fhP20fOq7CZCPMpPGBPXoB0dW0y46G6_4iHYouqejcWtpdRNzRYyq4VBCayqjeM__UnCngh3SUd4WA9q0F5-KtwBeTUVA_eWVjf63QwYb5uCZOL6JzX-a5nviPMksuBUr3-ye4yxXxGQtcv-smnfXGOhQFkfY";
    };
    heartbeatInterval = 120;
    httpRootProperty = locations;
    httpTimeout = 60000;
    iOSHasWarnedLocationServicesOff = 0;
    isFirstBoot = 0;
    isMoving = 0;
    lastLocationAuthorizationStatus = 3;
    locationAuthorizationAlert = {
        cancelButton = Cancel;
        instructions = "To use background location, you must enable '{locationAuthorizationRequest}' in the Location Services settings";
        settingsButton = Settings;
        titleWhenNotEnabled = "Background location is not enabled";
        titleWhenOff = "Location services are off";
    };
    locationAuthorizationRequest = Always;
    locationTemplate = "";
    locationTimeout = 60;
    locationsOrderDirection = ASC;
    logLevel = 5;
    logMaxDays = 3;
    maxBatchSize = 20;
    maxDaysToPersist = 3;
    maxRecordsToPersist = "-1";
    method = POST;
    minimumActivityRecognitionConfidence = 70;
    odometer = "4832.175934938634";
    params = {
    };
    pausesLocationUpdatesAutomatically = 1;
    persistMode = 2;
    preventSuspend = 1;
    schedule = (
        "2 9:00-23:00",
        "6 11:15-11:35",
        "3 21:40-22:00",
        "5 9:00-18:00",
        "4 8:30-17:41"
    );
    schedulerEnabled = 1;
    showsBackgroundLocationIndicator = 1;
    startOnBoot = 1;
    stationaryRadius = 50;
    stopAfterElapsedMinutes = "-1";
    stopDetectionDelay = 0;
    stopOnStationary = 0;
    stopOnTerminate = 0;
    stopTimeout = 2;
    trackingMode = 1;
    url = "https://api.mapsly.com/location_tracking/save";
    useSignificantChangesOnly = 0;
}

There were 2 walks further 500 meters: 14.09.2021 12:40-22:00 and 15.09.2021 10:55-11:35 . As you can see there are no logs during this periods. It worth to mention that device had no internet connection during the periods

We remember about your advice to increase stopTimeout parameter (it on test now). But could the stopTimeout be the only reason for this silence?
Or does the plugin requires internet connection to work?
Also should we create a separate issue for this case?
[background-geolocation (10).log.gz](https://github.com/transistorsoft/cordova-background-geolocation-lt/files/7169989/background-geolocation.10.log.gz)

**Comment by christocracy** [comment]
Please learn to use [Github Syntax Highlighting](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) on your code-blocks.

**Comment by christocracy** [comment]
>Also should we create a separate issue for this case?

This issue is titled "*Ask for Motion & Fitness via requestPermission() on ios*".  Yes, open a new issue for unrelated problems.
